### ISSUE TITLE
Runnable tasks should probably not be seen in fenced code blocks

### ISSUE THREAD

**Issue by anderseknert** [zed-industries/zed#12688]
### Description

After getting runnable tasks to work, I'm now seeing them also in markdown files where my language (`rego`) is embedded. While it would be cool to run tests inside of markdown files, OPA doesn't currently provide a way to do that, and pressing the play button will only result in errors. Is there some way to make the runnables conditional based on file extension, or whatever else makes the most sense to differentiate this context from when editing a regular file? I can get the `$ZED_FILE` from the task, but is there some way to make the task conditional based on that?

<img width="661" alt="Screenshot 2024-06-05 at 16 28 41" src="https://github.com/zed-industries/zed/assets/510711/5d80682a-0377-4157-9c3b-17f467a7a3e6">

(not sure if this is a bug or a feature request, so feel free to re-label as appropriate)

### Environment

Zed: v0.137.6 (Zed)
OS: macOS 14.5.0
Memory: 64 GiB
Architecture: aarch64

**Comment by anderseknert** [comment]
Yes, this is still an issue.

<img width="322" alt="Image" src="https://github.com/user-attachments/assets/6c133148-e2e3-40a1-a058-ef6e7d64b9fc" />

**Comment by anderseknert** [comment]
Still relevant.

**Comment by miguelraz** [comment]
Sorry for the noise, the bot got an issue upstream and went haywire: https://github.com/actions/stale/issues/1302.

**Comment by anderseknert** [comment]
Oh 😄 All good, thanks for letting me know!

**Comment by deadnews** [comment]
This also affects language extensions that inject bash into task runner files:

- **Makefiles**: [caius/zed-make#30](https://github.com/caius/zed-make/issues/30) — every recipe line gets a spurious run button (each `shell_text` node is a separate injection → separate `(program)` root)
- **Justfiles**: [jackTabsCode/zed-just#39](https://github.com/jackTabsCode/zed-just/issues/39) — first command line per recipe gets a run button (entire `recipe_body` is one injection → one `(program)` root)

Both are caused by the bash `runnables.scm` firing on injected content. Neither extension can work around this — there's no `injection.skip-runnables` or similar opt-out mechanism.

**Comment by deadnews** [comment]
```
Version: 1.8.2+stable
```

Same.

**Comment by anderseknert** [comment]
Yes. Bugs are either fixed or they’re not, they almost never get “outdated” just by virtue of moving fast.

**Comment by amaralenzo** [comment]
reproduced on zed 1.12.0+stable.328 (f96212f), windows 11 10.0.26200. still happening on latest stable and with built-in languages, not just extension ones

made a markdown file with rust/bash/go/python fences plus one fence with no language tag as a control:

- run markers show up in the editor gutter inside all tagged fences
- the control fence with no language tag has the same rust content and shows nothing
- a real control.rs in the same project shows markers normally, so runnables themselves are fine
- clicking the bash marker opens a terminal with a task named `run '...[my-file-name-truncated].md'` so i think it tries to run the whole markdown file as a shell script and it ends up just an empty terminal, nothing happens
- clicking the rust/go/python markers does literally nothing

so it's the embedded language support inside the fences generating the buttons, even when the language is just embedded in markdown for highlighting, which lines up with what @deadnews found on the makefile/justfile injections

<img width="1942" height="1102" alt="Image" src="https://github.com/user-attachments/assets/2f53dbb8-83bf-4522-8b94-cd646be4ab16" />

<img width="1917" height="1077" alt="Image" src="https://github.com/user-attachments/assets/9f82dddb-7b06-4a69-9e1c-96c3773a09e6" />

 @yeskunall  @AidanV
